[
    {
        "thread_id": 140,
        "name": "psyvenrix",
        "poster_url": "https://bitcointalk.org/index.php?action=profile;u=619",
        "subject": "Having problems specifing -datadir",
        "url": "https://bitcointalk.org/index.php?topic=601.msg6188#msg6188",
        "date": "July 28, 2010, 01:38:00 PM",
        "content": "<div class=\"post\">Whilst not amazingly critical, i find it strange that i consistently get the below error messages from the console and debug.log when i try to specify the datadir option with anything other than the period char (current directory).<br/>This is using the bitcoind program on an archlinux terminal session, im not using the GUI at all (headless box accessed via ssh)<br/><br/>From Console:<br/>************************<br/>EXCEPTION: 22DbRunRecoveryException<br/>DbEnv::open: DB_RUNRECOVERY: Fatal error, run database recovery<br/>bitcoin in AppInit()<br/><br/>terminate called after throwing an instance of 'DbRunRecoveryException'<br/>\u00a0\u00a0what(): \u00a0DbEnv::open: DB_RUNRECOVERY: Fatal error, run database recovery<br/><br/>and from the debug.log inside the datadir : -<br/><br/>Bitcoin version 0.3.4.0 beta<br/>Default data directory /home/psyvenrix/.bitcoin<br/>Bound to port 8333<br/>Loading addresses...<br/>dbenv.open strLogDir=./datadir2/database strErrorFile=./datadir2/db.log<br/><br/><br/>************************<br/>EXCEPTION: 22DbRunRecoveryException<br/>DbEnv::open: DB_RUNRECOVERY: Fatal error, run database recovery<br/>bitcoin in AppInit()<br/><br/>Yes, that is a SVN build, as the tar.gz from the main site is linked against openssl ver 0.9.8 i think and my arch linux system has 1.0.0 i think, so i just grabed the latest SVN source and built that (aside from a makefile tweak it compiled ok).<br/><br/>ive checked the permissions on both the directory containing the binaries and the aforementioned datadir, all are writable by my login (&amp; no other ACL/SELinux trickery in the background either)<br/><br/>I have only just started using bitcoin, so i find it strange that it's throwing a recovery type error when the contents of that directory are freshly generated (i.e. i can rm -rf the lot, then start bitcoind with a new non-existent directory, watch it create the files needed then bomb out)<br/><br/>Thoughts anyone? <img alt=\"Huh\" border=\"0\" src=\"https://bitcointalk.org/Smileys/default/huh.gif\"/></div>",
        "post_num": 1,
        "is_displayed": true,
        "nested_level": 0
    },
    {
        "thread_id": 140,
        "name": "Gavin Andresen",
        "poster_url": "https://bitcointalk.org/index.php?action=profile;u=224",
        "subject": "Re: Having problems specifing -datadir",
        "url": "https://bitcointalk.org/index.php?topic=601.msg6190#msg6190",
        "date": "July 28, 2010, 01:40:57 PM",
        "content": "<div class=\"post\">Does it work if you specify the full path?\u00a0 e.g. -datadir=/home/psyvenrix/datadir2\u00a0 ?<br/></div>",
        "post_num": 2,
        "is_displayed": true,
        "nested_level": 0
    },
    {
        "thread_id": 140,
        "name": "satoshi",
        "poster_url": "https://bitcointalk.org/index.php?action=profile;u=3",
        "subject": "Re: Having problems specifing -datadir",
        "url": "https://bitcointalk.org/index.php?topic=601.msg6268#msg6268",
        "date": "July 28, 2010, 08:58:26 PM",
        "content": "<div class=\"post\">It was able to reproduce this.\u00a0 The database doesn't like the relative path.<br/><br/>\"bitcoind -datadir=./subdir getinfo\" works against a running daemon, but trying to start the daemon as \"bitcoind -datadir=./subdir\" gets that exception.<br/><br/>I guess we should resolve the full path before passing it to the database.<br/><br/>It looks like you were the first one to ever use -datadir with a relative path.</div>",
        "post_num": 3,
        "is_displayed": true,
        "nested_level": 0
    },
    {
        "thread_id": 140,
        "name": "Tilka",
        "poster_url": "https://bitcointalk.org/index.php?action=profile;u=687",
        "subject": "Re: Having problems specifing -datadir",
        "url": "https://bitcointalk.org/index.php?topic=601.msg7644#msg7644",
        "date": "August 05, 2010, 07:07:36 AM",
        "content": "<div class=\"post\">+1 for quick bug fix. I'd like to be able to do that, too.</div>",
        "post_num": 4,
        "is_displayed": true,
        "nested_level": 0
    },
    {
        "thread_id": 140,
        "name": "lachesis",
        "poster_url": "https://bitcointalk.org/index.php?action=profile;u=237",
        "subject": "Re: Having problems specifing -datadir",
        "url": "https://bitcointalk.org/index.php?topic=601.msg8773#msg8773",
        "date": "August 11, 2010, 05:58:43 PM",
        "content": "<div class=\"post\"><div class=\"quoteheader\"><a href=\"https://bitcointalk.org/index.php?topic=601.msg6268#msg6268\">Quote from: satoshi on July 28, 2010, 08:58:26 PM</a></div><div class=\"quote\">It was able to reproduce this.\u00a0 The database doesn't like the relative path.<br/><br/>\"bitcoind -datadir=./subdir getinfo\" works against a running daemon, but trying to start the daemon as \"bitcoind -datadir=./subdir\" gets that exception.<br/><br/>I guess we should resolve the full path before passing it to the database.<br/><br/>It looks like you were the first one to ever use -datadir with a relative path.<br/></div>I've tried to do it before too, but I never considered it a large enough bug to report. It would be nice to see a fix, though.</div>",
        "post_num": 5,
        "is_displayed": true,
        "nested_level": 0
    },
    {
        "thread_id": 140,
        "name": "satoshi",
        "poster_url": "https://bitcointalk.org/index.php?action=profile;u=3",
        "subject": "Re: Having problems specifing -datadir",
        "url": "https://bitcointalk.org/index.php?topic=601.msg8924#msg8924",
        "date": "August 12, 2010, 09:43:29 PM",
        "content": "<div class=\"post\">Fixed in SVN rev 130.</div>",
        "post_num": 6,
        "is_displayed": true,
        "nested_level": 0
    }
]